src/lib/mongoBackend/mongoGetSubscriptions.cpp

```diff
@@ -260,14 +262,20 @@ void mongoGetSubscription
   const std::string&                  tenant
 )
 {
-  bool  reqSemTaken = false;
+  bool         reqSemTaken = false;
+  std::string  err;
+
+  if ((err = idCheck(idSub)) != "OK")
```

Comment:
I would suggest to use safeGetSubId() in the checking at mongoBackend layer due to two reasons:

1. Homogeneity (similar checkings in other parts of mongoBackend are using that method).
2. Using a different method to the one used at parsing stage. This kind of "HA" will mean that is some fail occur in the checking done by idCheck(), then safeGetSubId() could catch (or the opossite).